### PR TITLE
Add inline to cache flags

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: base/loading.jl [hunk 1]
```diff
@@ -2505,26 +2505,33 @@ function check_clone_targets(clone_targets)
     end
 end
 
+const OPT_LEVEL = 5
```
Comment (KristofferC) [inline base/loading.jl:2508]: Why is this a global?

Comment (vchuravy) [reply base/loading.jl:2508]: I wanted to mimick https://github.com/JuliaLang/julia/pull/48179/files#diff-1adc5c0f4e183651111760a5b2fdaf21418217de8142791cf1214709ae49dbc6R615

but yeah this isn't scoped.

File: base/loading.jl [hunk 2]
```diff
@@ -2505,26 +2505,33 @@ function check_clone_targets(clone_targets)
...
-    CacheFlags(f::Int) = CacheFlags(UInt8(f))
     function CacheFlags(f::UInt8)
         use_pkgimages = Bool(f & 1)
         debug_level = Int((f >> 1) & 3)
-        check_bounds = Bool((f >> 2) & 1)
-        opt_level = Int((f >> 4) & 3)
-        new(use_pkgimages, debug_level, check_bounds, opt_level)
+        check_bounds = Bool((f >> 3) & 1)
+        inline = Bool((f >> 3) & 1)
+        opt_level = Int((f >> OPT_LEVEL) & 3)
```
Comment (vchuravy) [inline base/loading.jl:2523]:
```suggestion
        opt_level = Int((f >> 5) & 3) # define OPT_LEVEL in statiddata_utils
```

File: base/loading.jl [hunk 3]
```diff
@@ -2505,26 +2505,33 @@ function check_clone_targets(clone_targets)
     end
 end
 
+const OPT_LEVEL = 5
+
```
Comment (vchuravy) [inline base/loading.jl:2509]:
```suggestion
```

File: base/loading.jl [hunk 4]
```diff
@@ -2506,25 +2506,30 @@ function check_clone_targets(clone_targets)
...
 
-    CacheFlags(f::Int) = CacheFlags(UInt8(f))
     function CacheFlags(f::UInt8)
         use_pkgimages = Bool(f & 1)
         debug_level = Int((f >> 1) & 3)
-        check_bounds = Bool((f >> 2) & 1)
-        opt_level = Int((f >> 4) & 3)
-        new(use_pkgimages, debug_level, check_bounds, opt_level)
+        check_bounds = Bool((f >> 3) & 1)
+        inline = Bool((f >> 3) & 1)
```
Comment (KristofferC) [inline base/loading.jl:2520]:
```suggestion
        inline = Bool((f >> 4) & 1)
```

?

File: src/staticdata_utils.c [hunk 5]
```diff
@@ -613,17 +613,18 @@ static void write_mod_list(ios_t *s, jl_array_t *a)
...
-    flags |= (jl_options.use_pkgimages & 1);
-    flags |= (jl_options.debug_level & 3) << 1;
-    flags |= (jl_options.check_bounds & 1) << 2;
-    flags |= (jl_options.opt_level & 3) << 4;
-    // NOTES:
-    // In contrast to check-bounds, inline has no "observable effect"
-    // CacheFlags in loading.jl should be kept in-sync with this
+    flags |= (jl_options.use_pkgimages & 1); // 0-bit
+    flags |= (jl_options.debug_level & 3) << 1; // 1-2 bit
+    flags |= (jl_options.check_bounds & 3) << 3; // 3-4 bit
```
Comment (vchuravy) [inline src/staticdata_utils.c:625]: So many fun bugs...

File: base/options.jl [hunk 6]
```diff
@@ -21,7 +21,6 @@ struct JLOptions
     startupfile::Int8
     compile_enabled::Int8
     code_coverage::Int8
-    malloc_log::Int8
     tracked_path::Ptr{UInt8}
```
Comment (vchuravy) [inline base/options.jl:24]:
```suggestion
    malloc_log::Int8
    tracked_path::Ptr{UInt8}
```

File: base/loading.jl [hunk 7]
```diff
@@ -2527,25 +2527,30 @@ function check_clone_targets(clone_targets)
 end
 
 struct CacheFlags
-    # ??OOCDDP - see jl_cache_flags
+    # OOICCDDP - see jl_cache_flags
     use_pkgimages::Bool
     debug_level::Int
     check_bounds::Bool
```
Comment (vchuravy) [inline base/loading.jl:2533]:
```suggestion
    check_bounds::Int
```

File: base/loading.jl [hunk 8]
```diff
@@ -2527,25 +2527,30 @@ function check_clone_targets(clone_targets)
...
     opt_level::Int
 
-    CacheFlags(f::Int) = CacheFlags(UInt8(f))
     function CacheFlags(f::UInt8)
         use_pkgimages = Bool(f & 1)
         debug_level = Int((f >> 1) & 3)
-        check_bounds = Bool((f >> 2) & 1)
-        opt_level = Int((f >> 4) & 3)
-        new(use_pkgimages, debug_level, check_bounds, opt_level)
+        check_bounds = Bool((f >> 3) & 3)
```
Comment (vchuravy) [inline base/loading.jl:2540]:
```suggestion
        check_bounds = Int((f >> 3) & 3)
```

**Review by IanButterworth — APPROVED** [review]

**Pull request merged**